app/models/article.rb

```diff
@@ -51,8 +51,8 @@ class Article < ApplicationRecord
 
   validates :body_markdown, length: { minimum: 0, allow_nil: false }, uniqueness: { scope: %i[user_id title] }
   validates :cached_tag_list, length: { maximum: 126 }
-  validates :canonical_url, url: { allow_blank: true, no_local: true,
-                                   schemes: %w[https http] }, uniqueness: { allow_blank: true }
+  validates :canonical_url, uniqueness: { allow_nil: true }
+  validates :canonical_url, url: { allow_blank: true, no_local: true, schemes: %w[https http] }
```

Comment:
split the validators in 2 lines and replaced `allow_blank` with `allow_nil` to avoid the empty string ;)